src/core/lib/iomgr/timer_generic.cc

```diff
@@ -256,7 +256,7 @@ static grpc_millis compute_min_deadline(timer_shard* shard) {
 static void timer_list_init() {
   uint32_t i;
 
-  g_num_shards = GPR_MIN(1, 2 * gpr_cpu_num_cores());
+  g_num_shards = GPR_MIN(32, 2 * gpr_cpu_num_cores());
```

Comment:
I think the original intention was to do `GPR_MAX(1, 2 * gpr_cpu_num_cores())`  (because on some platforms gpr_cpu_num_cores() may return 0).. But I like your change better :).

Reply:
Thank you Sree for the background! 

Reading your comment, I changed this to CLAMP(2xcores, 1, 32) to make sure it's never. PTAL. thanks!